### DISCUSSION TITLE
Slider questions:  changing min/max range and limiting events

### DISCUSSION THREAD

**Discussion by njk42** [ArthurSonzogni/FTXUI#455] [General]
Two part question related to Sliders:

1)  Is there any way to change the min, max, and increment on a Slider after its original creation?  Seems like these arguments should be passed by-pointer instead of by-value, so that later program flow can update these like it can for the value argument.

2)  After decorating a Slider with a CatchEvent(), some events (e.g. ArrowLeft and ArrowRight) are caught only when the focus is on that Slider element, but other events (e.g. mouse().button ) are caught regardless of where the mouse was clicked in the terminal window.  Is it possible to catch mouse button presses ONLY when the button is pressed in the Slider area and not elsewhere in the terminal?  If necessary, what is the recommended logic to use for querying the coordinates of the Slider and then sorting clicks based on mouse cursor position?

**Comment by ArthurSonzogni** [comment]
1. This sounds like a good idea! This would require updating the `int` arguments by the `ConstRef<int>` in the function signature. Not sure how this would interact with the existing template.

2. You can create a `Box` and render the element with a `reflect(&box)`. This way, the area used by the element will be reflected into `box`. Then you can check when getting a mouse event if this is in the box:
```cpp
  bool MyCustomComponent::OnEvent(Event event) final {
    if (!event.is_mouse()) {
      // [...]
      return false;
    }
    if (box_.Contain(event.mouse().x, event.mouse().y)) {
      // [...]
      return true;
    }
    return false;
```

```cpp
  Element MyCustomComponent::Render() override {
    return slider->Render() | reflect(box_);
  }

```

**Comment by njk42** [comment]
Thanks.  Ultimately just giving the Slider class a corresponding SliderOptions class with an .on_change member would be awesome to round out the functionality of this component.
